opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/InterfaceToNodeCache.java

```diff
@@ -48,8 +56,49 @@
 	 */
 	void clear();
 
-	Optional<Integer> getFirstNodeId(String location, InetAddress ipAddr);
+	default Optional<Entry> getFirst(String location, InetAddress ipAddr) {
+		final Iterator<Entry> it = this.get(location, ipAddr).iterator();
+		if (it.hasNext()) {
+			return Optional.of(it.next());
+		} else {
+			return Optional.empty();
+		}
+	}
+
+	default Optional<Integer> getFirstNodeId(String location, InetAddress ipAddr) {
+		return this.getFirst(location, ipAddr).map(e -> e.nodeId);
+	}
 
 	void removeInterfacesForNode(int nodeId);
 
+	class Entry {
+		public final int nodeId;
+		public final int interfaceId;
+
+		public Entry(final int nodeId, final int interfaceId) {
+			this.nodeId = nodeId;
+			this.interfaceId = interfaceId;
+		}
+
+		@Override
+		public boolean equals(final Object o) {
+			if (this == o) {
+				return true;
+			}
+
+			if (!(o instanceof Entry)) {
+				return false;
+
+			}
+			final Entry entry = (Entry) o;
+			return this.nodeId == entry.nodeId &&
+				   this.interfaceId == entry.interfaceId;
+		}
+
+		@Override
+		public int hashCode() {
+			return Objects.hash(this.nodeId,
+								this.interfaceId);
```

Comment:
Is it just looking so weird here or is it formatted this way?

Reply:
tabs vs spaces

(╯°□°)╯︵ ┻━┻